target/linux/ar71xx/image/senao.mk

```diff
@@ -0,0 +1,30 @@
+define Build/senao-factory-image
+	mkdir -p $(KDIR_TMP)/$(1)
+	touch $(KDIR_TMP)/$(1)/FWINFO-OpenWrt-$(REVISION)-$(1)
+	# $(CP) ./$(1)/* $(KDIR_TMP)/$(1)
+	dd if=$(KDIR)/$(1)-kernel.bin \
+		of=$(KDIR_TMP)/$(1)/openwrt-senao-$(1)-uImage-lzma.bin bs=64k conv=sync
+	dd if=$(KDIR)/root.squashfs \
+		of=$(KDIR_TMP)/$(1)/openwrt-senao-$(1)-root.squashfs bs=64k conv=sync
+	( \
+		cd $(KDIR_TMP)/$(1);  \
+		$(TAR) -cz -f $(call factoryname,squashfs,$(1)) * \
+	)
+	rm -rf $(KDIR_TMP)/$(1)
+endef
+
+define Device/ens202ext
```

Comment:
Please have a closer look at the code! The factory image still has two separate files. It is just that openwrt-senao-$(1)-uImage-lzma.bin is the kernel + as much of the rootfs that fits into the 1536k kernel partition that is used by the stock firmware. openwrt-senao-$(1)-root.squashfs contains the remaining bits of the rootfs. This way the oem kernel + rootfs partition (== firmware in LEDE) has the content expected by the LEDE firmware partition splitter.